### DISCUSSION TITLE
Request access to an email 📧

### DISCUSSION THREAD

**Discussion by Bonapara** [twentyhq/twenty#1615] [Ideas]
Emails should appear differently to users, according to the specific rights they have

**Table of content:**
1. Full visibility
2. Object and metadata
3. Metadata
4. How to request access to an email

# Full visibility

When the email recipients have set their privacy settings to "Everything", or have been granted access to the email (see below), the email content should be fully visible. In both cases the email should be clickable:

![Image](https://github.com/twentyhq/twenty/assets/19412894/c2961fea-3a52-490a-8697-c5a34e5a8216)

### Email tab

![Image](https://github.com/twentyhq/twenty/assets/19412894/0e97b80f-d6f8-44eb-a7b2-73cfc2fe02ad)

### Timeline

![Image](https://github.com/twentyhq/twenty/assets/19412894/a9b6f5d4-823e-450c-aada-7d9724ea08e4)

# Object & metadata visibility

When the email recipients have set their privacy settings to "Object and Metadata"

![Image](https://github.com/twentyhq/twenty/assets/19412894/b5a0a226-11a2-4c34-8539-e3f332517e75)

### Email tab

**Fake content (lorem ipsum)** is displayed under a blur layer, and a "require access" button is visible on hover. The email is not clickable

https://github.com/twentyhq/twenty/assets/19412894/40ae93a5-7487-42d8-9752-d9e3f5d5892f

### Timeline

![Image](https://github.com/twentyhq/twenty/assets/19412894/7a09f8ee-c45d-4d13-866e-e402cc34b568)

# Metadata-only visibility

When recipients have set their privacy settings to "Metadata"

![Image](https://github.com/twentyhq/twenty/assets/19412894/ec73eba2-8d1d-42ad-a4d1-99cbd1228b13)

### Email tab

We don't display the object, and the **fake content** should take all the width.

https://github.com/twentyhq/twenty/assets/19412894/68bc7865-5cb8-4f4e-8e4a-90786a7c80cd

### Timeline

a fake object should be displayed blur alongside the blurred fake content:

![Image](https://github.com/twentyhq/twenty/assets/19412894/a1d6709f-6ee6-47d6-b18e-12ee76381002)

# Request access 

The purpose of this feature is to notify a team member when someone else tries to access an email they wrote to a company. So, if one team member contacts a company and another team member later tries to view that email, the original sender gets a notification.

This notification should appear in the notification center, stating someone has required access to an email. only **one** notification should appear (even if the user has clicked several times on "require access"). If the recipient has refused to give access in the notification center, then the user can require access again, and a new notification should be sent. (We'll add limits later)

Archive, should "ignore" (say no) meaning a user can request access again

https://github.com/twentyhq/twenty/assets/19412894/440d0326-31e6-49e2-a719-8cd8d0f8d1fd

**Comment by FelixMalfait** [comment]
Data model: **MessageThreadWorkspaceMember**
id, createdAt, updatedAt
messageThreadId - string fk
workspaceMemberId - nullable, fk to users

**Comment by BOHEUS** [comment]
@Bonapara is request access planned to be added once messages are surfaced and notifications are implemented? Rest of discussion with specific rights is already implemented

**Reply by FelixMalfait** [reply]
I feel like request to access is a topic that should be tackled as part of permissions/roles. It will be the same underlying logic as "share a note" or "share a task". So it's a big workstream, that needs to be rethough from first principles not just emails

**Comment by Bonapara** [comment]
Closing this in favor of https://github.com/twentyhq/twenty/discussions/18895.

Per Felix's comment here:
https://github.com/twentyhq/twenty/discussions/1615#discussioncomment-16227341

the remaining request-access topic should be handled as part of a broader permissions / roles workstream rather than as an email-only feature. Continuing the discussion there so the model can cover emails, notes, tasks, and similar resources consistently.
